corehq/apps/translations/forms.py

```diff
@@ -316,13 +316,28 @@ def clean(self):
         cleaned_data = super(AddTransifexBlacklistForm, self).clean()
         app_id = cleaned_data.get('app_id')
         module_id = cleaned_data.get('module_id')
-
-        app_json = Application.get_db().get(app_id)
-        for module in app_json['modules']:
-            if module_id == module['unique_id']:
-                break
-        else:
-            raise forms.ValidationError("Module {} not found in app {}".format(module_id, app_json['name']))
+        field_type = cleaned_data.get('field_type')
+
+        self._check_module_in_app(app_id, module_id)
+        self._check_module_not_ui_translation(module_id, field_type)
+        self._check_module_for_case_list_detail(module_id, field_type)
+
+    def _check_module_in_app(self, app_id, module_id):
+        if module_id:
+            app_json = Application.get_db().get(app_id)
+            for module in app_json['modules']:
+                if module_id == module['unique_id']:
+                    break
+            else:
+                raise forms.ValidationError("Module {} not found in app {}".format(module_id, app_json['name']))
+
+    def _check_module_not_ui_translation(self, module_id, field_type):
+        if module_id and field_type == 'ui':
+            raise forms.ValidationError("Leave Module ID blank for UI translations")
+
+    def _check_module_for_case_list_detail(self, module_id, field_type):
+        if not module_id and field_type != 'ui':
+            raise forms.ValidationError("Module ID must be given for Case List or Case Detail")
```

Comment:
You could also use `self.add_error` in some of these, rather than raising a validation error.  That'll allow you to attach the error to a specific field in the form.

Reply:
Nice! Thank you.